casual/casual-api/src/main/java/se/laz/casual/api/util/work/RepeatUntilSuccessTaskWork.java

```diff
@@ -20,20 +23,26 @@ public class RepeatUntilSuccessTaskWork<T> implements Work
     private final Supplier<T> supplier;
     private final Consumer<T> consumer;
     private final Supplier<WorkManager> workManagerSupplier;
+    private int failures = 0;
+    private final long maxBackoffMillis;
+    ScheduledExecutorService scheduledExecutorService = Executors.newSingleThreadScheduledExecutor();
```

Comment:
Can you instead utilise the inbuilt `ManagedScheduleExecutorService`? Or does this not work in both weblogic and wildfly?

Reply:
I remember having problems looking up the `ManagedScheduledExecutorService` from jndi on WildFly and can't just `@Resource` it in in this context so I used this instead to achieve something that works. I can have another quick look at why it failed and if I can manage to use `ManagedScheduledExecutorService` in a portable manner

Reply:
wls has extensions to the JCA specification - one of them being that CDI works.
It is not in the specification and thus will never work on wildfly unless they also implement an extension which I doubt that they will.

Reply:
After looking at it some more: on WildFly I can't just lookup `java:comp/DefaultManagedScheduledExecutorService`, maybe because I'm not in an EJB context currently or something, I don't know. What does yield a result is to lookup the jboss default `java:jboss/ee/concurrency/scheduler/default` which gives me a `ManagedScheduledExecutorService` that works. Seems we already have some sort of adjacent hack around getting our hands a `ManagedExecutorService` via https://github.com/casualcore/casual-java/blob/f913f5c0dccd5fea47bfaad1537c02675636353a/casual/casual-network/src/main/java/se/laz/casual/network/outbound/JEEConcurrencyFactory.java#L28

Would it really be worth it to extend the hack to be able to fetch `ManagedScheduledExecutorService` instances as well?

Reply:
The problem being that we do not know the direct names for WLS - thus that only works for wildfly.
I remember that the these direct names are actually documented by wildfly, they are not by wls.

Which is pretty much the comment in that file I see :)

Reply:
Since we now have a split between 3.2 and 2.2 - we could say that 2.2 is wls only going forward.
That would allow us to use CDI stuff on 2.2 and the direct names on 3.2.
If we want to do that, @cklellie @TobiasLeo ?